packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -206,18 +206,17 @@ export default class CashierStore extends BaseStore {
         // eslint-disable-next-line no-undef
         reaction(
             () => [
-                this.root_store.client.switched,
+                this.root_store.client.loginid,
```

Comment:
`switched` is a control flag we use to detect account changes, this shouldn't be the issue. 

```js
@action.bound
async switchAccount(loginid) {
    this.setPreSwitchAccount(true);
    this.setIsLoggingIn(true);
    this.root_store.ui.removeNotifications(true);
    this.root_store.ui.removeAllNotificationMessages(true);
    this.setSwitched(loginid);
    this.responsePayoutCurrencies(await WS.authorized.payoutCurrencies());
}```
Is there a way to keep that, or the `switched` flag is buggy? The reason I'm asking this is that other stores also rely on this, and maybe we need to investigate more if they have such an issue.

Reply:
In this PR, it seems that the issue is our cashier-store `init` reacts to `this.root_store.client.switched` too early before  `is_virtual` property inside `client-store` is updated, so it leads to incorrect evaluation of `is_virtual` property that hasn't been changed yet at that moment. When we changed `switched` property to `loginid`, it looks like it reacts to it at the right moment when client-store `is_virtual` value is updated, so we do not face any issues with it.
---
If we still want to keep `switched` property, we would have to manually call `get_settings` call and force `reaction` method to wait for the response which I think makes the system slower by some time and not the best solution. You can refer to my previous commit to see how it looks like.
---
I'm not sure if it's a bug of `switched` setter but looks like it happens because `switched` is changed at different time from `loginid` and we have this asynchronous issues with client-store.